src/Psl/DateTime/Internal/format_rfc3339.php

```diff
@@ -0,0 +1,54 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Psl\DateTime\Internal;
+
+use Psl\DateTime\SecondsStyle;
+use Psl\DateTime\Timestamp;
+use Psl\DateTime\Timezone;
+
+use function substr;
+
+/**
+ * @internal
+ *
+ * @psalm-mutation-free
+ *
+ * @psalm-suppress ImpureMethodCall
+ */
+function format_rfc3339(
+    Timestamp $timestamp,
+    ?SecondsStyle $seconds_style = null,
+    bool $use_z = false,
+    null|Timezone $timezone = null,
+): string {
+    $seconds = $timestamp->getSeconds();
+    $nanoseconds = $timestamp->getNanoseconds();
+
+    if (null === $seconds_style) {
+        $seconds_style = match (true) {
```

Comment:
could make sense to move this match expression to a static constructor on SecondsStyle maybe?

Reply:
makes sense 👍